src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -3104,5 +3039,64 @@ static string Print(Range? arg)
         }
     }
 }";
+
+        [Fact]
+        [WorkItem(54085, "https://github.com/dotnet/roslyn/issues/54085")]
+        public void OrderOfEvaluation_01()
+        {
+            var comp = CreateCompilationWithIndexAndRangeAndSpan(@"
+using System;
+
+class CollectionX
+{
+    private int[] _array = new[] { 1, 2, 3 };
+
+    public int Length
+    {
+        get
+        {
+            Console.WriteLine(""Length"");
+            return _array.Length;
+        }
+    }
+
+    public ref int this[int index]
+    {
+        get
+        {
+            Console.WriteLine(""Indexer get"");
+            return ref _array[index];
+        }
+    }
+}
+
+static class SideEffect
+{
+    static CollectionX Get()
+    {
+        Console.WriteLine(""Get"");
+        return new CollectionX();
+    }
+    static int GetIdx()
+    {
+        Console.WriteLine(""GetIdx"");
+        return 1;
+    }
+
+    public static void Main()
+    {
+        int i2 =  Get()[Index.FromEnd(GetIdx())];
```

Comment:
Is there any value in having a version of the test which uses `Get()[^GetIdx()]`?